temporalio/worker/_nexus.py

```diff
@@ -0,0 +1,467 @@
+                        operation_token=result.token,
+                        links=links,
+                    )
+                )
+            elif isinstance(result, nexusrpc.handler.StartOperationResultSync):
+                [payload] = await self._data_converter.encode([result.value])
+                return temporalio.api.nexus.v1.StartOperationResponse(
+                    sync_success=temporalio.api.nexus.v1.StartOperationResponse.Sync(
+                        payload=payload,
+                        links=links,
+                    )
+                )
+            else:
+                raise _exception_to_handler_error(
+                    TypeError(
+                        "Operation start method must return either "
+                        "nexusrpc.handler.StartOperationResultSync or "
+                        "nexusrpc.handler.StartOperationResultAsync."
+                    )
+                )
+        except nexusrpc.OperationError as err:
+            return temporalio.api.nexus.v1.StartOperationResponse(
+                operation_error=await self._operation_error_to_proto(err),
+            )
+
+    async def _exception_to_failure_proto(
+        self,
+        err: BaseException,
+    ) -> temporalio.api.nexus.v1.Failure:
+        try:
+            api_failure = temporalio.api.failure.v1.Failure()
+            await self._data_converter.encode_failure(err, api_failure)
+            _api_failure = google.protobuf.json_format.MessageToDict(api_failure)
+            return temporalio.api.nexus.v1.Failure(
+                message=_api_failure.pop("message", ""),
+                metadata={"type": "temporal.api.failure.v1.Failure"},
+                details=json.dumps(_api_failure).encode("utf-8"),
+            )
+        except BaseException as err:
+            return temporalio.api.nexus.v1.Failure(
+                message=f"{err.__class__.__name__}: {err}",
+                metadata={"type": "temporal.api.failure.v1.Failure"},
+            )
+
+    async def _operation_error_to_proto(
+        self,
+        err: nexusrpc.OperationError,
+    ) -> temporalio.api.nexus.v1.UnsuccessfulOperationError:
+        return temporalio.api.nexus.v1.UnsuccessfulOperationError(
+            operation_state=err.state.value,
+            failure=await self._exception_to_failure_proto(err),
+        )
+
+    async def _handler_error_to_proto(
+        self, err: nexusrpc.HandlerError
+    ) -> temporalio.api.nexus.v1.HandlerError:
+        return temporalio.api.nexus.v1.HandlerError(
+            error_type=err.type.value,
+            failure=await self._exception_to_failure_proto(err),
+            retry_behavior=(
+                temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_RETRYABLE
+                if err.retryable
+                else temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_NON_RETRYABLE
```

Comment:
This is wrong, `retryable` can be `None` in which case it would translate to unspecified.